cmd/npmExecuteLint_test.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"errors"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/stretchr/testify/assert"
+	"path/filepath"
+	"testing"
+)
+
+type mockLintUtilsBundle struct {
+	*mock.FilesMock
+	execRunner *mock.ExecMockRunner
+}
+
+func (u *mockLintUtilsBundle) getExecRunner() execRunner {
+	return u.execRunner
+}
+
+func (u *mockLintUtilsBundle) getGeneralPurposeConfig(configURL string) {
+	u.AddFile(filepath.Join(".pipeline", ".eslintrc.json"), []byte(`abc`))
+}
+
+func newLintMockUtilsBundle() mockLintUtilsBundle {
+	utils := mockLintUtilsBundle{FilesMock: &mock.FilesMock{}, execRunner: &mock.ExecMockRunner{}}
+	return utils
+}
+
+func TestNpmExecuteLint(t *testing.T) {
+	t.Run("Call with ci-lint script and one package.json", func(t *testing.T) {
+		lintUtils := newLintMockUtilsBundle()
+		lintUtils.AddFile("package.json", []byte("{\"scripts\": { \"ci-lint\": \"\" } }"))
+
+		npmUtils := newNpmMockUtilsBundle()
+		npmUtils.execRunner = lintUtils.execRunner
+
+		config := npmExecuteLintOptions{}
+
+		npmExecutor := npmExecutorMock{utils: npmUtils, config: npmConfig{runScripts: []string{"ci-lint"}, runOptions: []string{"--silent"}}}
```

Comment:
Are runScripts and runOptions still required?

Reply:
Ah, this is part of the assertion, right?

Reply:
Yes, since the `npmConfig` struct holds the expected parameters for the mocking implementation to check if the functions were called with the correct parameters.

When only mocking the file system calls this is not required anymore cf: https://github.com/SAP/jenkins-library/blob/stepNpmLint/cmd/npmExecuteLint_test.go#L123